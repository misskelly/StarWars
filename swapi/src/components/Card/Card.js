import React from 'react'

const personCard = ({name, homeworld, species, popOfHome}) => {
return (
  <article className='card'>
    <h3>{name}</h3>
    <ul>
      <li>species: {species}</li>
      <li>homeworld: {homeworld}</li>
      <li>population: {popOfHome}</li>
    </ul>
    <button>Favorite</button>
  </article>
)
}

export default personCard;

// Name
// Homeworld
// Species
// Population of Homeworld
// A button to“ Favorite” the person