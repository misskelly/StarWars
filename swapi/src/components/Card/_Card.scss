.card {
  display: inline-block;
  transition: all 0.3s ease;
  border-radius: 5px;
  border: 1px solid hsl(20, 3%, 54%);
  box-shadow: 3px 2px 2px hsl(200, 100%, 1%);
  background: hsla(221, 96%, 48%, 0.351);
  min-width: 250px;
  position: relative;
  padding: 1rem;
  color: hsl(0, 0%, 84%);
}

.card:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  box-shadow: 5px 5px 5px hsl(204, 25%, 20%);
}

.card-heading {
  padding: 0 0 1rem 0;
  margin-bottom: 1rem;
  text-align: center;
  font-size: 1.5rem;
  font-family: $alien;
  border-bottom: 1px solid $yellow;
}

.fav-btn {
  float: right;
  position: relative;
  margin: 1rem 0 0 0;
  cursor: pointer;

  &:hover {
    background-color: transparent;
    color: $yellow;
  }
}