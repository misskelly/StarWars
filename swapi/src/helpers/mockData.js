export const mockCleanFilm = {
     scroll: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     year: "1999"
   }
export const mockDirtyFilm = {
     opening_crawl: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     release_date: "1999-07-03"
   }


