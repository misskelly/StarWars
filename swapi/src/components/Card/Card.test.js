import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';
import { mockCleanPeople } from '../../helpers/mockData.js';


describe('Card', () => {
      let wrapper;
      let index;
      let category;
      let item;

  beforeEach(() => {
    index = 0;
    category = 'people';
    item = mockCleanPeople[0];
    wrapper = shallow(<Card key={`${index}_${item.name}`} item={item} category={category}/>)
  })

  it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

})