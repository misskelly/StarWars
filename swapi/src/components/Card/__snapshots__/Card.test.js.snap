// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<article
  className="card"
>
  <h3
    className="card-heading"
  >
    Luke Skywalker
  </h3>
  <ul>
    <li
      className="card-detail"
    >
      species: 
      Human
    </li>
    <li
      className="card-detail"
    >
      homeworld: 
      Tatooine
    </li>
    <li
      className="card-detail"
    >
      population: 
      200000
    </li>
  </ul>
  <button
    className="fav-btn"
  >
    Favorite
  </button>
</article>
`;
